plugins/amd-pmc/amd_pmc_test.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: LGPL-2.1+
+
+import json
+import os
+import sys
+import unittest
+from fwupd_test import FwupdTest
+
+
+class AmdPmcTest(FwupdTest):
+    def setUp(self):
+        super().setUp()
+        self.testbed.add_from_string(
+            """P: /devices/platform/AMDI0009:00
+E: DRIVER=amd_pmc
+E: ID_PATH=platform-AMDI0009:00
+E: ID_PATH_TAG=platform-AMDI0009_00
+E: ID_VENDOR_FROM_DATABASE=Amdek Corporation
+E: MODALIAS=acpi:AMDI0009:PNP0D80:
+E: SUBSYSTEM=platform
+L: driver=../../../bus/platform/drivers/amd_pmc
+A: driver_override=(null)
+L: firmware_node=../../LNXSYSTM:00/LNXSYBUS:00/AMDI0009:00
+A: modalias=acpi:AMDI0009:PNP0D80:
+A: power/control=auto
+A: power/runtime_active_time=0
+A: power/runtime_status=unsupported
+A: power/runtime_suspended_time=0
+A: smu_fw_version=76.78.0
+A: smu_program=0
+"""
+        )
+
+    def test_amd_pmc_device(self):
+        """Verify the AMD PMC device is detected correctly"""
+        self.start_daemon()
+        cmd = self.fwupdmgr_command("get-devices")
```

Comment:
dumb question; why not use the GObjectIntrospection support of libfwupd? seems less hack that exec-ing a process and then deserialzing json.

Reply:
Hmm for most intents and purposes that should work pretty well.